core/modules/qdisp/QueryResource.cc

```diff
@@ -44,46 +44,53 @@
 #include "qdisp/JobStatus.h"
 #include "qdisp/QueryRequest.h"
 #include "qdisp/QueryResource.h"
-#include "qdisp/ResponseRequester.h"
 
 namespace lsst {
 namespace qserv {
 namespace qdisp {
 
+QueryResource::QueryResource(std::shared_ptr<JobQuery> jobQuery)
+      // this char* must live as long as this object, so copy it on heap
+    : Resource(::strdup(jobQuery->getDescription().resource().path().c_str())),
+      _xrdSsiSession(NULL), _jobQuery(jobQuery) {
+    if (rName == NULL) {
```

Comment:
I haven't upgraded my compiler, I was going to after this ticket and either clean this up as part of DM-2722 or put in another ticket to do the more minor things. Also, that's why I'm using NULL and not 0.